packages/asset-uploader/src/db/clone.ts

```diff
@@ -0,0 +1,53 @@
+import { prisma, type Prisma, type Project } from "@webstudio-is/prisma-client";
+import {
+  authorizeProject,
+  type AppContext,
+} from "@webstudio-is/trpc-interface/server";
+
+export const cloneAssets = async (
+  {
+    fromProjectId,
+    toProjectId,
+    dontCheckEditPermission = false,
+  }: {
+    fromProjectId: Project["id"];
+    toProjectId: Project["id"];
+
+    /**
+     * Don't use unless absolutely have to (e.g. because of transactions)
+     * and unless it's obvious on the call site that permission is checked
+     */
+    dontCheckEditPermission?: boolean;
```

Comment:
why would you need this flag instead of just having the right permissions for the action?

Reply:
See further in diff where we call `cloneAssets`: https://github.com/webstudio-is/webstudio-designer/pull/969/files#diff-3130be2fe9639006f0fd0765501c1b1415bfb9952e34ea773bcb0400a02e22efR230

`hasProjectPermit()` returns `false` until project is created in DB